### DISCUSSION TITLE
maskClosable属性的默认值支持全局配置

### DISCUSSION THREAD

**Discussion by bbcvc** [ant-design/ant-design#50831] [RFCs]
想法：点击蒙层的交互行为通常是全局统一的，而不是每个modal表现出不一样的行为;
之前  https://github.com/ant-design/ant-design/issues/47995  有提到过对组件二次封装，但这样实际上还是会增加维护与团队沟通上的成本，使用这些组件时可能并意识不到需要从别的模块导入

建议maskClosable属性支持全局配置
```js
function Demo() {
  return (
     <ConfigProvider maskClosable={{ Modal: false, Drawer: false }}>
          <App/>
     </ConfigProvider>
  )
}
```
类似的：
https://github.com/ant-design/ant-design/issues/28042
https://github.com/ant-design/ant-design/issues/47995

**Comment by Wxh16144** [comment]
全是老生常谈的问题，感觉都要一个一个暴露也很麻烦.. 这次 mask 下次又会有 keyboard 全局配置..

**Reply by bbcvc** [reply]
这种具有全局配置特征的属性的默认值，能不能通过**某种模式得到普遍的支持**，既能得到有效支持也可以降低antd升级的成本
1. 解决这种升级还要在局部考虑默认的设置兼容问题
![image](https://github.com/user-attachments/assets/86200fc5-de08-41af-a332-e126bc375286)
2. 默认的设置可能不符合antd使用时，在面对不同业务的场景需求，更加灵活

设想可以不可以类似于 **FYI**
```js
const Modal: React.FC<ModalProps> = (props) => {
    // 从全局拿默认值，而不是写死在组件内部
    const globalConfig = useGlobalConfig('Modal')
    const defauleMaskClosable = globalConfig.maskClosable
    const mergeProps= useMerge(props, {
       maskClosable: defauleMaskClosable
    })
    
    return wrapCSSVar(
      <ContextIsolator form space>
            <Dialog
               {...mergeProps}
               ...
             />
       </ContextIsolator>
    )
}
```

**Reply by bbcvc** [reply]
@afc163  能支持吗？
